lib/driver.js

```diff
@@ -1,39 +1,55 @@
-﻿import { BaseDriver } from 'appium-base-driver';
+﻿import _ from 'lodash';
+import { BaseDriver } from 'appium-base-driver';
 import { system } from 'appium-support';
 import WinAppDriver from './winappdriver';
 import logger from './logger';
 
-const WINAPPDRIVER_PORT = 4723;
+const WINAPPDRIVER_PORT = 5000;
 const WINAPPDRIVER_HOST = "localhost";
 
 // Appium instantiates this class
 class WindowsDriver extends BaseDriver {
   constructor (opts = {}, shouldValidateCaps = true) {
     super(opts, shouldValidateCaps);
     this.jwpProxyActive = false;
-    this.opts.port = opts.port || WINAPPDRIVER_PORT;
     this.opts.address = opts.address || WINAPPDRIVER_HOST;
   }
 
-  async createSession (caps) {
+  async createSession (caps, reqCaps, curSessions) {
+
     if (!system.isWindows()) {
       throw new Error("WinAppDriver tests only run on Windows");
     }
     try {
       let sessionId;
       [sessionId] = await super.createSession(caps);
-      await this.startWinAppDriverSession();
+      await this.startWinAppDriverSession(curSessions);
       return [sessionId, caps];
     } catch (e) {
       await this.deleteSession();
       throw e;
     }
   }
 
-  async startWinAppDriverSession () {
+  getNextAvailablePort(curSessions) {
+    let newWADPort = WINAPPDRIVER_PORT;
+
+    // start at 5000 and go up till we find a port that isn't in use
+    while (_.find(curSessions, function(o) {return o.WADPort == newWADPort})) {
```

Comment:
we now use the es6 style in something like this:

```js
while (_.find(curSessions, (o) => o.WADPort === newWADPort)) {
```
(note also the `===` instead of `==`